code/modules/lighting/lighting_setup.dm

```diff
@@ -1,16 +1,12 @@
-/proc/create_all_lighting_overlays()
-	for(var/zlevel = 1 to world.maxz)
-		create_lighting_overlays_zlevel(zlevel)
-
-/proc/create_lighting_overlays_zlevel(var/zlevel)
-	ASSERT(zlevel)
-
-	for(var/turf/T in block(locate(1, 1, zlevel), locate(world.maxx, world.maxy, zlevel)))
-		if(!T.dynamic_lighting)
+/proc/create_all_lighting_objects()
+	for(var/area/A in GLOB.all_areas)
```

Comment:
Something in here is causing the lighting subsystem to take 35 seconds to initialize. If anyone has any insight here, I'd love to hear it. This works the same as it does on /tg/, so I'm a bit puzzled.

Reply:
Shitloads of areas? Maybe we should do `for(var/i in GLOB.all_areas)` then under that do `var/area/a = i` to see if that speeds it up a tiny bit- and for the turfloop below

We've seen on occasion funky shiz like that can yield slight performance improvement